### ISSUE TITLE
Cancel current builds if a dependency is changed during a build

### ISSUE THREAD

**Issue by jazzdan** [windmilleng/tilt#1816]
If I have a build going, especially if it takes a long time, and I change a dependency that triggers that same resource to build the current build should be cancelled so that the new build can be started.

_Reported by @dotellie in #tilt_

**Comment by jazzdan** [comment]
Coming back to this issue with some real world experience ... the current Tilt behavior causes a lot of folks to disable automatic builds. After they're done writing some code, and they've done a bunch of saves, they switch to the Tilt window and are unsure if the builds that Tilt are building is for their current change, or if once this build finishes another build will need to finish before they can see all of their changes take effect. As a result they'd rather do manual builds and now what is going to be in it, and have things potentially take a longer, than save as many seconds as possible.

In other words: it seems like people value predictability over raw speed.

Our builds take 15s-60s FWIW.
